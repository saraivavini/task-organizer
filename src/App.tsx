import React from 'react';
import { NativeBaseProvider } from 'native-base';
import { Routes } from './routes';
import { ThemeProvider } from 'styled-components/native';
import './locales';
import { theme } from './global/theme';

export default function App() {
  return (
    <NativeBaseProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <Routes />
      </ThemeProvider>
    </NativeBaseProvider>
  );
}
