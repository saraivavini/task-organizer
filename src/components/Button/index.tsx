import { Button as NBButton, Text } from 'native-base';

export type ButtonProps = {
  label: string;
  onPress: () => void;
};

export const Button = (props: ButtonProps) => {
  const { label, onPress } = props;

  return (
    <NBButton
      backgroundColor="brand.primary"
      size="lg"
      borderRadius="8px"
      onPress={onPress}
    >
      <Text fontSize="lg" fontWeight="medium" color="white">
        {label}
      </Text>
    </NBButton>
  );
};
