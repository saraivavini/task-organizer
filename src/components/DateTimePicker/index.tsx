import RNDateTimePicker, {
  DateTimePickerEvent,
} from '@react-native-community/datetimepicker';
import { Box, Pressable, Text, Modal } from 'native-base';
import { useState } from 'react';
import { Platform } from 'react-native';
import { DateHandler } from '../../helpers';
import { Input } from '../Input';

type DateTimePickerProps = {
  type?: 'date' | 'time';
  label?: string | null;
  onChange?: (date: Date) => void;
};

const formatByTypeMapping = {
  date: 'DAY_MONTH_YEAR',
  time: 'HOUR_WITH_MINUTES',
} as const;

const iconByTypeMapping = {
  date: 'calendar',
  time: 'clock',
} as const;

export const DateTimePicker = (props: DateTimePickerProps) => {
  const { type = 'date', label, onChange = () => null } = props;

  const [date, setDate] = useState(new Date());
  const [isVisible, setIsVisible] = useState(false);

  const formattedDate = DateHandler.formatDate(date, formatByTypeMapping[type]);

  const handleChange = (_: DateTimePickerEvent, newDate?: Date) => {
    if (newDate) {
      setDate(newDate);

      setIsVisible(Platform.OS === 'ios');
    }
  };

  const toggleVisible = () => {
    setIsVisible(!isVisible);
    onChange(date);
  };

  const display = Platform.OS === 'ios' ? 'spinner' : 'default';

  return (
    <Box>
      <Input
        label={label}
        isReadOnly
        value={formattedDate}
        icon={iconByTypeMapping[type]}
        onPress={toggleVisible}
      />
      <Modal isOpen={isVisible} onClose={toggleVisible}>
        <Modal.Content>
          <RNDateTimePicker
            display={display}
            value={date}
            mode={type}
            onChange={handleChange}
            is24Hour
          />
        </Modal.Content>
      </Modal>
    </Box>
  );
};
