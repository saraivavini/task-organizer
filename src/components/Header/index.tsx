import { Box } from 'native-base';
import { Logo } from '../Logo';

export const Header = () => {
  return (
    <Box mb={6}>
      <Logo size="small" />
    </Box>
  );
};
