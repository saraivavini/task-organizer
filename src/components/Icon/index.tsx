import { Icon as NBIcon, IIconProps as NBIconProps } from 'native-base';
import { FeatherIcon } from '../../@types';
import Feather from '@expo/vector-icons/Feather';

type IconProps = {
  testID?: string;
  name: FeatherIcon;
} & Pick<NBIconProps, 'color' | 'size'>;

export const Icon = (props: IconProps) => {
  const { name, size, color = 'muted.400', testID } = props;

  return (
    <NBIcon
      as={Feather}
      testID={testID}
      color={color}
      name={name}
      size={size}
    />
  );
};
