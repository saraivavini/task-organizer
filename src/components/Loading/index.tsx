import { Modal, Spinner } from 'native-base';

type LoadingProps = {
  isVisible: boolean;
};

export const Loading = ({ isVisible }: LoadingProps) => {
  return (
    <Modal isOpen={isVisible}>
      <Spinner size="lg" colorScheme={'brand'} />
    </Modal>
  );
};
