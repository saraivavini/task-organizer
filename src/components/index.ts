export * from './Button';
export * from './DateTimePicker';
export * from './Header';
export * from './Icon';
export * from './IconButton';
export * from './Input';
export * from './Logo';
export * from './ScreenContainer';
export * from './Loading';
