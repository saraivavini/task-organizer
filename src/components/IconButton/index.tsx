import { Pressable } from 'native-base';
import { IColors } from 'native-base/lib/typescript/theme/base/colors';
import { FeatherIcon } from '../../@types';
import { Icon } from '../Icon';

type IconButtonProps = {
  testID?: string;
  icon: FeatherIcon;
  color?: IColors;
  onPress: () => void;
  size?: number;
};

export const IconButton = (props: IconButtonProps) => {
  const { icon, onPress, color = 'muted.400', size = 6, testID } = props;

  return (
    <Pressable onPress={onPress} testID={testID}>
      <Icon size={size} color={color} name={icon} />
    </Pressable>
  );
};
