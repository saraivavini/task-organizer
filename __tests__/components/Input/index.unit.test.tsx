import { Input } from '../../../src/components';
import { componentBuilder } from '../../utils/componentBuilder';

const { renderComponent } = componentBuilder(Input, {
  onChange: jest.fn(),
  value: '',
  placeholder: '',
  type: 'text',
});

describe('Input tests', () => {
  it('Should renders correctly', () => {
    const { component } = renderComponent();

    expect(component.toJSON()).toMatchSnapshot();
  });
});
