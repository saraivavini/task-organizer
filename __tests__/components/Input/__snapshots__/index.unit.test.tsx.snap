// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests Should renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "width": "100%",
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <View
        dataSet={Object {}}
        isFocused={false}
        style={
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 8,
            "borderColor": "#d4d4d4",
            "borderRadius": 4,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 8,
            "borderWidth": 1,
            "flex": 1,
            "flexDirection": "row",
            "marginLeft": 0,
            "overflow": "hidden",
          }
        }
      >
        <TextInput
          accessible={true}
          dataSet={Object {}}
          editable={true}
          onBlur={[Function]}
          onChangeText={[MockFunction]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          placeholder=""
          placeholderTextColor="#a3a3a3"
          secureTextEntry={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderBottomLeftRadius": 0,
              "borderTopLeftRadius": 0,
              "color": "#171717",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "400",
              "height": "100%",
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
              "width": "100%",
            }
          }
          value=""
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
